src/coreComponents/physicsSolvers/wavePropagation/ElasticWaveEquationSEM.cpp

```diff
@@ -0,0 +1,746 @@
+
+  registerWrapper( viewKeyStruct::receiverIsLocalString(), &m_receiverIsLocal ).
+    setInputFlag( InputFlags::FALSE ).
+    setSizedFromParent( 0 ).
+    setDescription( "Flag that indicates whether the receiver is local to this MPI rank" );
+
+  registerWrapper( viewKeyStruct::displacementxNp1AtReceiversString(), &m_displacementxNp1AtReceivers ).
+    setInputFlag( InputFlags::FALSE ).
+    setSizedFromParent( 0 ).
+    setDescription( "Displacement value at each receiver for each timestep (x-component)" );
+
+  registerWrapper( viewKeyStruct::displacementyNp1AtReceiversString(), &m_displacementyNp1AtReceivers ).
+    setInputFlag( InputFlags::FALSE ).
+    setSizedFromParent( 0 ).
+    setDescription( "Displacement value at each receiver for each timestep (y-component)" );
+
+  registerWrapper( viewKeyStruct::displacementzNp1AtReceiversString(), &m_displacementzNp1AtReceivers ).
+    setInputFlag( InputFlags::FALSE ).
+    setSizedFromParent( 0 ).
+    setDescription( "Displacement value at each receiver for each timestep (z-component)" );
+
+}
+
+ElasticWaveEquationSEM::~ElasticWaveEquationSEM()
+{
+  // TODO Auto-generated destructor stub
+}
+
+void ElasticWaveEquationSEM::initializePreSubGroups()
+{
+
+  WaveSolverBase::initializePreSubGroups();
+
+  DomainPartition & domain = this->getGroupByPath< DomainPartition >( "/Problem/domain" );
+
+  NumericalMethodsManager const & numericalMethodManager = domain.getNumericalMethodManager();
+
+  FiniteElementDiscretizationManager const &
+  feDiscretizationManager = numericalMethodManager.getFiniteElementDiscretizationManager();
+
+  FiniteElementDiscretization const * const
+  feDiscretization = feDiscretizationManager.getGroupPointer< FiniteElementDiscretization >( m_discretizationName );
+  GEOSX_THROW_IF( feDiscretization == nullptr,
+                  getName() << ": FE discretization not found: " << m_discretizationName,
+                  InputError );
+
+
+}
+
+
+void ElasticWaveEquationSEM::registerDataOnMesh( Group & meshBodies )
+{
+
+  meshBodies.forSubGroups< MeshBody >( [&]( MeshBody & meshBody )
+  {
+
+    MeshLevel & meshLevel =  meshBody.getMeshLevel( 0 );
+
+    NodeManager & nodeManager = meshLevel.getNodeManager();
+
+    nodeManager.registerExtrinsicData< extrinsicMeshData::Displacementx_nm1,
+                                       extrinsicMeshData::Displacementy_nm1,
+                                       extrinsicMeshData::Displacementz_nm1,
+                                       extrinsicMeshData::Displacementx_n,
+                                       extrinsicMeshData::Displacementy_n,
+                                       extrinsicMeshData::Displacementz_n,
+                                       extrinsicMeshData::Displacementx_np1,
+                                       extrinsicMeshData::Displacementy_np1,
+                                       extrinsicMeshData::Displacementz_np1,
+                                       extrinsicMeshData::ForcingRHS_x,
```

Comment:
I think for consistency in naming there should be no underscore before x, y or z (if @francoishamon agrees)